tests/components/garmin_connect/test_config_flow.py

```diff
@@ -21,98 +20,91 @@
 }
 
 
-@pytest.fixture(name="mock_garmin_connect")
-def mock_garmin():
-    """Mock Garmin Connect."""
-    with patch(
-        "homeassistant.components.garmin_connect.config_flow.Garmin",
-    ) as garmin:
-        garmin.return_value.login.return_value = MOCK_CONF[CONF_ID]
-        yield garmin.return_value
-
-
 async def test_show_form(hass):
     """Test that the form is served with no input."""
-
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["errors"] == {}
     assert result["step_id"] == config_entries.SOURCE_USER
 
 
 async def test_step_user(hass):
     """Test registering an integration and finishing flow works."""
-
     with patch(
-        "homeassistant.components.garmin_connect.Garmin.login",
-        return_value="my@email.address",
-    ), patch(
-        "homeassistant.components.garmin_connect.async_setup_entry", return_value=True
-    ):
+        "homeassistant.components.garmin_connect.config_flow.Garmin",
+    ) as garmin:
+        garmin.return_value.login.return_value = MOCK_CONF[CONF_ID]
         result = await hass.config_entries.flow.async_init(
             DOMAIN, context={"source": config_entries.SOURCE_USER}, data=MOCK_CONF
         )
     assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
     assert result["data"] == MOCK_CONF
 
 
-async def test_connection_error(hass, mock_garmin_connect):
+async def test_connection_error(hass):
     """Test for connection error."""
-    mock_garmin_connect.login.side_effect = GarminConnectConnectionError("errormsg")
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=MOCK_CONF
-    )
+    with patch(
+        "garminconnect_ha.Garmin.login",
```

Comment:
Make sure to patch the path where the name is used, not where it's defined.

https://docs.python.org/3/library/unittest.mock.html#where-to-patch